src/jsp/jsp_sr.c

```diff
@@ -269,11 +282,30 @@ delay_load_hook (unsigned dliNotify, PDelayLoadInfo pdli)
     {
     case dliFailLoadLib:
       {
-	char *java_home = NULL, *tmp = NULL, *tail;
-	char jvm_lib_path[BUF_SIZE];
-	void *libVM;
+	char *java_home = NULL, *jvm_path = NULL, *tmp = NULL, *tail;
+	void *libVM = NULL;
 
+	jvm_path = getenv ("JVM_PATH");
 	java_home = getenv ("JAVA_HOME");
+
+	if (jvm_path)
+	  {
+	    err_msgs.append ("\n\tFailed to load libjvm from 'JVM_PATH' envirnment variable: ");
```

Comment:
(1) envirnment -> environment
(2) is it right that error message append to err_msgs, if jvm_path is not null ?

Reply:
(1) Thank you for catching the typo.
(2) If it succeeds to load libjvm, err_msgs will not be used. But, for readability, It would better to move the code. I'll fix it.

Reply:
On windows, If an error occurred in the middle of calling LoadLibrary at the next, it jumps to the exception handler. To gather the appropriate error message, append the error first here.